src/components/records.tsx

```diff
@@ -1,40 +1,84 @@
 import React, { useEffect, useState } from "react"
 import theme from "../config/theme"
-import { GSheetsTable } from "../components/table"
+import { getSheetData } from "../api-calls"
+import Table from "../components/table"
 import Card from "../components/card"
 
 const SHEET_ID = "1-0FhSZemh9iVF5bEOnqRGIH3ZFAHb9-ktbw5I2Zz9eE"
 
+type Record = {
+  event: string
+  name: string
+  time: string
+  year: string
+  race: string
+  link?: string
+  current?: string
+}
+
 type RecordTables = {
-  men: JSX.Element
-  women: JSX.Element
+  men: Record[]
+  women: Record[]
 }
 
-function getRecords(raceType: string): RecordTables {
-  return {
-    men: getRecordsForGender(raceType, "Men"),
-    women: getRecordsForGender(raceType, "Women"),
-  }
+async function getRecords(raceType: string): RecordTables {
+  const records = [getRecordsForGender(raceType, "Men"), getRecordsForGender(raceType, "Women")];
+  const [men, women] = await Promise.all(records);
+  return { men, women }
 }
 
-function getRecordsForGender(raceType: string, gender: string): JSX.Element {
+async function getRecordsForGender(raceType: string, gender: string): JSX.Element {
   const sheetName = raceType + " - " + gender;
-  return GSheetsTable({ sheetId: SHEET_ID, range: sheetName, fontsize: 15, padding: 8 });
+  return await getSheetData(SHEET_ID, `${sheetName}!A2:Z100`)
+    .then((records) => records.map(record => ({
+      event: record[0],
+      name: record[1],
+      time: record[2],
+      year: record[3],
+      race: record[4],
+      link: record.length > 5 ? record[5] : null,
+      current: record.length > 6 ? record[6] : null
+    })));
```

Comment:
This seems brittle -- can we determine column indices based on the header row in the sheet?